### DISCUSSION TITLE
Unable to use Quill

### DISCUSSION THREAD

**Discussion by ghost** [odygrd/quill#464] [Q&A]
Hello everybody,
I was trying to make Quill work in my project but for some reason i have troubles with linkage, my OS is Windows.
I tried adding Quill via add_subdirectory as an embedded lib and via find_package after installing it but the thing is - it doesn't compile to begin with, there's no lib file produced with .a extension in the lib folder, when i try 'cmake --build .' it results in a message 'ninja: no work to do'. As I understand it should start compilation and produce a lib file but it doesn't.
When i add it via add_subdirectory command it configures but compilation ends with linker errors, where's the lib file to link against?. I'm VERY confused.

I created a test project without any other dependencies and still got linker errors like:
[build] [ 50%] Building CXX object CMakeFiles/test.dir/main.cpp.obj
[build] [100%] Linking CXX executable test.exe
[build] C:/mingw64/bin/ld: CMakeFiles\test.dir/objects.a(main.cpp.obj): in function `std::once_flag::_Prepare_execution::_Prepare_execution<std::call_once<quill::Backend::start(quill::BackendOptions const&)::{lambda()#1}>(std::once_flag&, quill::Backend::start(quill::BackendOptions const&)::{lambda()#1}&&)::{lambda()#1}>(quill::Backend::start(quill::BackendOptions const&)::{lambda()#1}&)':

and other gibberish ending in:
[build] clang++: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by ghost** [comment]
nevermind, i use clang and it seems to be the issue, changing kits to gcc helped
pls delete this discussion

**Reply by odygrd** [reply]
Hello, I have tested it with vs 2019 msvc conpiler and vs 2022 msvc there are also github actions here https://github.com/odygrd/quill/actions/runs/9373079011

It also works with the embedded mingw64 that comes with CLion.

I am not sure how mingw64 works with clang on windows as i work mainly on Linux. 

Out of curiosity what mingw64 and clang version are you using ?
Also which quill version are you using ? Version >= 4 is header only there won't be any libquill.a
